sdk/test-utils/identity-extensions/package.json

```diff
@@ -0,0 +1,77 @@
+{
+  "name": "@azure-tools/identity-extensions",
```

Comment:
We should avoid calling this identity-extensions. Extensions used to be the work-in-progress name of the plugin API, and we're not using the plugin API here. But also, we don't want someone to stumble on this utility package and think it's something for general use by identity users.

How about `@azure-tools/test-credential`?